src/QMCTools/QMCGaussianParserBase.cpp

```diff
@@ -475,8 +479,11 @@ xmlNodePtr QMCGaussianParserBase::createIonSet()
       hout.push(SpecieID.str().c_str(), true);
       int z          = static_cast<int>(ionSpecies(AtomicNumberIndex, i));
       double valence = ionSpecies(IonChargeIndex, i);
-      if (valence > CoreTable[z] && FixValence != true)
-        valence -= CoreTable[z];
+      if (z < gCoreTable.size())
+      {
+        if (valence > gCoreTable[z] && !FixValence)
+          valence -= gCoreTable[z];
+      }
```

Comment:
Please add else and throw error.

Reply:
@ye-luo a couple of things as it seems like a larger effort due to unintended consequences:
1. `converter_test_Bi_dirac` would trigger this error as `z` is going out of bounds (tests pass as gcc/clang are permissive and zero initialize memory for the most part). 
2. Throwing an exception generates leaks due to the skipped `xmlFree*` lines (mixing C++ and C)

Reply:
The original code is simply wrong if someone runs an element heavier than Xe.  We should error abort for z too large and print a meaningful error message.

Reply:
I think adding `FixValence = true;` to the DiracParser constructor would avoid this issue for the converter test. The valence info is read and set correctly in the ascii parsing of the dirac output. This is similar to the QPParser and LCAOHDFParser constructors

Reply:
I'll try @camelto2 suggestion to have at least a leaky exception (but no test doing it). Thank you all.